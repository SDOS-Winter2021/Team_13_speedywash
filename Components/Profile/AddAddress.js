/**
 * @module
 */
import React from 'react'
import { Text, View, Button, TextInput } from 'react-native'
import keys from "../../configs/KEYS"
import AsyncStorage from '@react-native-community/async-storage';
import * as firebase from 'firebase';
import { Image, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native';
import { TouchableOpacity } from 'react-native';
import { FontAwesome5, MaterialIcons, AntDesign } from '@expo/vector-icons';
import { Modal, Dimensions } from "react-native";


/**
 * Adds Address 
 * @param {Object} obj - An object
 * @param {module:globaltypes.User} obj.currentUser - current user profile
 * @param {function} obj.setCurrentUser - current user setter
 * @param {function} obj.setcurrentView - current view setter
 * @return {View} - React Component View
*/
function AddAddress({ currentUser, setcurrentUser, setcurrentView }) {
    const services = [
        {
            id: "1",
            name: "Add",
            action: (onPress) => {
                handleAdd();
            }
        },
        {
            id: "2",
            name: "Go Back",
            action:(onPress) => setcurrentView({screen: keys.screens.PROFILE, header: true, footer:true})
        }
    ]
    const _renderItem = (item) => {
        return <View style={styles.serviceListItem}
            key={item.id}>
            <TouchableOpacity
                onPress = {()=>item.action()} // to be added
                style={styles.listItemTouchable}>
                <View style={{ flex: 0.6 }}>
                    <Text style={styles.ServiceNameStyle}>{item.name}</Text>
                </View>
            </TouchableOpacity>
        </View>
    }
    function handleAdd(){
        
    }

    return (
        <View style={styles.homeScreen}>
            <Text
                style={styles.AddAddressTitle}>
                Add New Address
            </Text> 
            <TextInput style = {styles.input}
                underlineColorAndroid = "transparent"
                placeholder = "Address Name" 
            />
            <TextInput style = {styles.input}
                underlineColorAndroid = "transparent"
                placeholder = "Enter Address" 
            />
            <TextInput style = {styles.input}
                underlineColorAndroid = "transparent"
                placeholder = "Enter State" 
            />
            <TextInput style = {styles.input}
                underlineColorAndroid = "transparent"
                placeholder = "Enter Pincode" 
            />
            <SafeAreaView style={{ flex: 1, marginBottom: 15 }}>
                {
                    services.map(_renderItem)
                }
            </SafeAreaView>
            
        </View>
    )
}

const styles = StyleSheet.create({
    homeScreen: {
        backgroundColor: keys.colors.HOMEBACKGROUND,
        flex: 1,
        width: '100%'
    },
    AddAddressTitle: {
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 25,
        marginBottom: 10,
        color: keys.colors.MAIN,
        marginVertical: '2%'
    },
    input: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.185,
        marginHorizontal: '4%',
        marginVertical: '1%',
        borderRadius: 5,
        textAlign: 'center',
        fontSize: 20
     },
    submitButton:{
        flexDirection: 'row',
        justifyContent: 'space-around',
        backgroundColor: 'white',
        width: '46%',
        height: undefined,
        aspectRatio: 1 / 0.185,
        marginHorizontal: '4%',
        marginVertical: '1%',   
    },
    listItemTouchable: {
        
        flexDirection: 'row',
        justifyContent: "space-evenly"
    },
    ServiceNameStyle: {
        fontSize: 20,
        alignSelf: 'center',
        margin: '10%'
    },
    serviceListItem: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.2,
        marginHorizontal: '4%',
        marginVertical: '1.5%',
        borderRadius: 7
    },
    
});
export default AddAddress
