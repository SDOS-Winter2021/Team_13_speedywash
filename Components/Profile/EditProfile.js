/**
 * @module
 */
import React, { useEffect, useState } from 'react'
import { Text, View, Button, TextInput } from 'react-native'
import keys from "../../configs/KEYS"
import AsyncStorage from '@react-native-community/async-storage';
import * as firebase from 'firebase';
import { Image, StyleSheet, Alert } from 'react-native';
import { SafeAreaView } from 'react-native';
import { TouchableOpacity } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { setValue } from "../../configs/CacheManager";

/**
 * Allows user to edit his/her displayName, email
 * and upload his/her profile image. All the changes are also 
 * commited to database as well
 * @param {Object} obj - An object
 * @param {module:globaltypes.User} obj.currentUser - current user profile
 * @param {function} obj.setCurrentUser - current user setter
 * @param {function} obj.setcurrentView - current view setter
 * @return {View} - React Component View
*/
function EditProfile({ currentUser, setcurrentUser, setcurrentView }) {

    const [avatarName, setavatarname] = useState(currentUser.displayName);
    const [userDetails, setUserDetails] = useState({
        name: currentUser.displayName,
        email: currentUser.email
    });
    // const [img,setImg] = useState(currentUser.photoURL == "" ? require("../../assets/avatar.png") : currentUser.photoURL);
    const [img, setImg] = useState(currentUser.photoURL == "" ? "../../assets/avatar.png" : currentUser.photoURL);
    const [isPicked, setPicked] = useState(currentUser.photoURL == "" ? false : true);

    const pickImage = () => {
        ImagePicker.requestMediaLibraryPermissionsAsync().then((status) => {
            if (status.granted) {
                ImagePicker.launchImageLibraryAsync({
                    mediaTypes: ImagePicker.MediaTypeOptions.Images,
                    allowsEditing: true,
                    aspect: [4, 3],
                    quality: 1
                }).then((result) => {
                    if (!result.cancelled) {
                        setImg(result.uri)
                        fetch(result.uri).then((response) => {
                            response.blob().then((blob) => {
                                let ref = firebase.storage().ref().child(`profile/${currentUser.uid}`)
                                ref.put(blob).then(() => {
                                    firebase.storage().ref(`profile/${currentUser.uid}`).getDownloadURL().then((url) => {
                                        firebase.firestore().collection("users").doc(currentUser.uid)
                                            .update({ photoURL: url })
                                            .then(() => {
                                                Alert.alert("Success", "Image Uploaded Successfully")
                                                setcurrentUser({ ...currentUser, photoURL: url })
                                            })
                                    })
                                })
                            })
                        })
                        setPicked(true);
                    }
                })
            }
        });
    }
    function handleName(text) {
        setUserDetails(prevValue => {
            return {
                name: text,
                email: prevValue.email
            };
        });
    }
    function handleEmail(text) {
        setUserDetails(prevValue => {
            return {
                name: prevValue.name,
                email: text
            };
        });

    }
    function handleUpdates() {
        const newObj = {
            "displayName": userDetails.name,
            "email": userDetails.email,
        }
        firebase.firestore().collection("users").doc(currentUser.uid).update(newObj).then(() => {

            AsyncStorage.setItem(keys.storage.USER, JSON.stringify({ ...currentUser, ...newObj }))
                .then(() => {
                    setcurrentUser({ ...currentUser, ...newObj })
                    setavatarname(userDetails.name);
                    Alert.alert("Success", "Details Updated Successfully")
                })
        });
    }
    // console.log(img)
    function AvatarCard() {
        return (
            <View style={styles.card}>
                <View>
                    <Text style={styles.cardText}>
                        {avatarName}
                    </Text>
                    <Text style={styles.cardText}>
                        {currentUser.phoneNumber}
                    </Text>

                </View>
                {/* <Image resize Mode="contain" style={styles.cardImage} source={
                    {uri: currentUser.photoURL == "" ? require("../../assets/avatar.png") : currentUser.photoURL}
                    } /> */}
                {isPicked ? <Image resize Mode="contain" style={styles.cardImage} source={{ uri: img }} /> :
                    <Image resize Mode="contain" style={styles.cardImage} source={require("../../assets/avatar.png")} />}
            </View>);
    }
    const services = [
        {
            id: "1",
            name: "Upload Image",
            action: (onPress) => {
                pickImage();
            }
        },
        {
            id: "2",
            name: "Update",
            action: (onPress) => {
                handleUpdates();
            }
        },
        {
            id: "3",
            name: "Go Back",
            action: (onPress) => setcurrentView({ screen: keys.screens.PROFILE, header: true, footer: true })
        }
    ]
    const _renderItem = (item) => {
        return <View style={styles.submitButton}
            key={item.id}>
            <TouchableOpacity
                onPress={() => item.action()} // to be added
                style={styles.listItemTouchable}>
                <View style={{ flex: 0.6 }}>
                    <Text style={styles.ServiceNameStyle}>{item.name}</Text>
                </View>
            </TouchableOpacity>
        </View>
    }
    return (
        <View style={styles.homeScreen}>
            <AvatarCard />
            <Text
                style={styles.EditProfileTitle}>
                Edit Profile
            </Text>
            <TextInput style={styles.input}
                underlineColorAndroid="transparent"
                value={userDetails.name}
                onChangeText={(text) => handleName(text)}
            />
            <TextInput style={styles.input}
                underlineColorAndroid="transparent"
                value={userDetails.email}
                onChangeText={(text) => handleEmail(text)}
            />
            <SafeAreaView style={{ flex: 1, marginBottom: 15 }}>
                {
                    services.map(_renderItem)
                }
            </SafeAreaView>
        </View>
    )
}

const styles = StyleSheet.create({
    homeScreen: {
        backgroundColor: keys.colors.HOMEBACKGROUND,
        flex: 1,
        width: '100%'
    },
    EditProfileTitle: {
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 25,
        marginBottom: 10,
        color: keys.colors.MAIN
    },
    input: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.185,
        marginHorizontal: '4%',
        marginVertical: '1%',
        borderRadius: 5,
        textAlign: 'center',
        fontSize: 20
    },
    card: {
        backgroundColor: "white",
        width: '92%',
        height: "25%",
        borderStyle: 'solid',
        borderWidth: 1,
        marginHorizontal: '4%',
        marginVertical: '1.5%',
        borderRadius: 17,
        marginVertical: "2%",
        marginBottom: "8%",
        marginTop: "5%",
        flexDirection: 'row',
        justifyContent: "space-evenly",
        alignItems: 'center'
    },
    cardImage: {
        width: 120,
        height: 120,
        borderRadius: 150 / 2,
        overflow: "hidden",
        borderWidth: 3,
        borderColor: keys.colors.MAIN,
    },
    cardText: {
        textAlign: 'left',
        textAlignVertical: 'center',
        fontSize: 18,
        color: keys.colors.MAIN,
        padding: "2%",
        marginHorizontal: "2%",
    },
    submitButton: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.185,
        marginHorizontal: '4%',
        marginVertical: '1%',
        borderRadius: 5
    },
    listItemTouchable: {
        flexDirection: 'row',
        justifyContent: "space-evenly"
    },
    ServiceNameStyle: {
        fontSize: 20,
        alignSelf: 'center',
        margin: '10%'
    },
});
export default EditProfile