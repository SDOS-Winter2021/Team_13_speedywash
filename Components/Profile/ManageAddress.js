import React, {useState} from 'react'
import { Text, View, Button, TextInput } from 'react-native'
import keys from "../../configs/KEYS"
import AsyncStorage from '@react-native-community/async-storage';
import * as firebase from 'firebase';
import {StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native';
import { TouchableOpacity } from 'react-native';
import {MaterialIcons} from '@expo/vector-icons';

import { Alert } from 'react-native';

function ManageAddress({ currentUser, setcurrentUser, setcurrentView }) {
    const addressList = [
        {
            id: "1",
            name: "Home",
            state: "New Delhi",
            address: "G-3, Milap Nagar, Uttam Nagar",
            pincode: "110059"
    
        },
        {
            id: "2",
            name: "Work",
            state: "New Delhi",
            address: "Boys hostel, IIITD, Okhla,",
            pincode: "110020"
        },
        {
            id: "3",
            name: "Work",
            state: "New Delhi",
            address: "Boys hostel, IIITD, Okhla,",
            pincode: "110020"
        }
    ]
    const [addresses,setAddress] = useState(addressList);

    function deleteAddress(curr_id){
        if(addresses.length == 1){
            Alert.alert("Atleast one address required") ;
        }
        else{
            setAddress(addresses.filter((item) => (item.id != curr_id)))
        }
    }

    function _renderAddress(curr_address) {
        return (
            <View style={styles.card}>
                <View >
                    <Text style={styles.cardText}>
                        {curr_address.name}
                    </Text>
                    <Text style={styles.cardText}>
                        {curr_address.address}
                    </Text>
                    <Text style={styles.cardText}>
                        {curr_address.state}
                    </Text>
                    <Text style={styles.cardText}> 
                        {curr_address.pincode}
                    </Text>
                </View> 
                <TouchableOpacity style={styles.icons} onPress={()=>deleteAddress(curr_address.id)}>
                    <MaterialIcons style={styles.cardImage} name='delete' size={30} color={'grey'}/>  
                </TouchableOpacity>  
            </View>);
    }   

    return (
        <View style = {styles.homeScreen}>
            <Text
                style={styles.ManageAddressTitle}>
                Manage Address
            </Text> 
            <SafeAreaView style={{ flex: 1, marginBottom: 15 }}>
                {
                    addresses.map(_renderAddress)
                }
            </SafeAreaView>
            <SafeAreaView style={styles.container}>
                <View style={styles.submitButton}>
                    <Button onPress={()=>setcurrentView({screen: keys.screens.PROFILE, header: true, footer: true})} title="Go Back"/>
                    <Button onPress={()=> addresses.length == 3 ? Alert.alert("Address limit reached") : 
                        setcurrentView({screen: keys.screens.ADDADDRESS, header: true, footer: true})} 
                        title="Add Address" />
                </View>
            </SafeAreaView>
        </View>
    );
}

const styles = StyleSheet.create({
    homeScreen: {
        backgroundColor: keys.colors.HOMEBACKGROUND,
        flex: 1,
        width: '100%'
    },
    ManageAddressTitle: {
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 25,
        marginBottom: 10,
        color: keys.colors.MAIN,
        marginVertical: '2%'
    },

    icons: {
        alignSelf: 'center'
    },

     card: {
        backgroundColor: "white",
        width: '92%',
        height: "25%",
        borderStyle: 'solid',
        borderWidth: 1,
        marginHorizontal: '4%',
        borderRadius: 15,
        marginBottom: "2%",
        marginTop: "2%",
        flexDirection: 'row',
        alignSelf: 'center',
        alignItems: 'center',
        
    },
    cardImage: {
        alignSelf : 'flex-end',  
        marginBottom: '4%',
        justifyContent : 'flex-end'
    },
    cardText: {
        textAlign: 'left',
        textAlignVertical: 'center',
        fontSize: 18,
        color: keys.colors.MAIN,
        padding: "1%",
        paddingHorizontal: '5%',
        marginHorizontal: "1%",
        
    },
    submitButton:{
        flexDirection: 'row',
        justifyContent: 'space-evenly',
        width: '46%',
        height: undefined,
        aspectRatio: 1 / 0.185,
        fontSize:18,
        marginHorizontal: '4%',
        marginVertical: '1%',
         
    },
});
export default ManageAddress
