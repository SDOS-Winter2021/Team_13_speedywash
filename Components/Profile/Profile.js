import React from 'react'
import { Text, View, Button } from 'react-native'
import keys from "../../configs/KEYS"
import AsyncStorage from '@react-native-community/async-storage';
import * as firebase from 'firebase';
import { Image, StyleSheet } from 'react-native';
import { SafeAreaView, ScrollView } from 'react-native';
import { TouchableOpacity } from 'react-native';
import { Avatar } from 'react-native-elements';
import { cleanCache } from "../../configs/CacheManager";

function Temporary({ currentUser, setcurrentUser }) {
    function performSignout() {
        firebase.auth().signOut()
        AsyncStorage.removeItem(keys.storage.USER).then(() => {
            setcurrentUser(null);
        })
    }
    return <View>
        <Text>{currentUser.displayName}</Text>
        <Text>{currentUser.email}</Text>
        <Text>{currentUser.phoneNumber}</Text>
        <Button title="Sign out" onPress={performSignout} />
    </View>
}

function Profile({ currentUser, setcurrentUser, setcurrentView }) {
    function AvatarCard() {
        return (
            <View style={styles.card}>
                <View>
                    <Text style={styles.cardText}>
                        {currentUser.displayName}
                    </Text>
                    <Text style={styles.cardText}>
                        {currentUser.phoneNumber}
                    </Text>
                </View>
                {currentUser.photoURL == "" ? <Image resize Mode="contain" style={styles.cardImage} source={require("../../assets/avatar.png")} /> : <Image resize Mode="contain" style={styles.cardImage} source={{ uri: currentUser.photoURL }} />}
            </View>);
    }

    const services = [
        {
            id: "1",
            name: "Edit Profile",
            action: (onPress) => setcurrentView({ screen: keys.screens.EDITPROFILE, header: true, footer: true })
        },
        {
            id: "2",
            name: "Subscription",
            action: () => { }
        },
        {
            id: "3",
            name: "My Orders",
            action: (onPress) => { setcurrentView({ screen: keys.screens.MYORDERS, header: true, footer: true})}
        },
        {
            id: "4",
            name: "Manage Address",
            action: (onPress) => setcurrentView({ screen: keys.screens.MANAGEADDRESS, header: true, footer: true })
        },
        {
            id: "5",
            name: "Sign Out",
            action: () => {
                firebase.auth().signOut()
                AsyncStorage.removeItem(keys.storage.USER).then(() => {
                    setcurrentUser(null);
                })
            }
        },
        {
            id: "6",
            name: "Clean Cache",
            action: () => {
                cleanCache();
            }
        }
    ];

    const _renderItem = (item) => {
        return <View style={styles.serviceListItem}
            key={item.id}>
            <TouchableOpacity
                onPress={() => item.action()} // to be added
                style={styles.listItemTouchable}>
                <View style={{ flex: 0.6 }}>
                    <Text style={styles.ServiceNameStyle}>{item.name}</Text>
                </View>
            </TouchableOpacity>
        </View>
    }

    return (
        <ScrollView 
        nestedScrollEnabled={false}
        showsVerticalScrollIndicator={false}
        style={styles.homeScreen}>
            <AvatarCard />
            <SafeAreaView style={{ flex: 1, marginBottom: "20%" }}>
                {
                    services.map(_renderItem)
                }
            </SafeAreaView>
        </ScrollView>
    )
}
const styles = StyleSheet.create({
    homeScreen: {
        backgroundColor: keys.colors.HOMEBACKGROUND,
        flex: 1,
        width: '100%'
    },
    card: {
        backgroundColor: "white",
        width: '92%',
        height: "25%",
        borderStyle: 'solid',
        borderWidth: 1,

        marginHorizontal: '4%',
        marginVertical: '1.5%',
        borderRadius: 17,
        marginVertical: "2%",
        marginBottom: "8%",
        marginTop: "5%",
        flexDirection: 'row',
        justifyContent: "space-evenly",
        alignItems: 'center'

    },
    cardImage: {
        // alignSelf : 'flex-end',
        width: 120,
        height: 120,
        borderRadius: 150 / 2,
        overflow: "hidden",
        borderWidth: 3,
        borderColor: keys.colors.MAIN,
    },
    cardText: {
        textAlign: 'left',
        textAlignVertical: 'center',
        fontSize: 18,
        color: keys.colors.MAIN,
        padding: "2%",
        marginHorizontal: "2%",
    },
    serviceListItem: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.2,
        marginHorizontal: '4%',
        marginVertical: '1.5%',
        borderRadius: 7
    },
    listItemTouchable: {
        flexDirection: 'row',
        justifyContent: "space-evenly"
    },
    ServiceNameStyle: {
        fontSize: 20,
        alignSelf: 'center',
        margin: '10%'
    },
    signOutStyle: {
        backgroundColor: 'white',
        width: '92%',
        height: undefined,
        aspectRatio: 1 / 0.2,
        marginHorizontal: '4%',
        marginVertical: '1.5%',
        borderRadius: 7
    }

});
export default Profile
