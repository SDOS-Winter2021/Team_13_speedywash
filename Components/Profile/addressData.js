export const addressList = [
    {
        id: "1",
        name: "Home",
        state: "New Delhi",
        address: "G-3, Milap Nagar, Uttam Nagar",
        pincode: "110059"

    },
    {
        id: "2",
        name: "Work",
        state: "New Delhi",
        address: "Boys hostel, IIITD, Okhla,",
        pincode: "110020"
    },
    {
        id: "3",
        name: "Work",
        state: "New Delhi",
        address: "Boys hostel, IIITD, Okhla,",
        pincode: "110020"
    }
]