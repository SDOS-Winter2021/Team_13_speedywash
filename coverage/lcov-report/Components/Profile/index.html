
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Components/Profile</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> Components/Profile</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">29.31% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>17/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>4/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">16.67% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>6/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">29.31% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>17/58</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="EditProfile.js"><a href="EditProfile.js.html">EditProfile.js</a></td>
	<td data-value="26.83" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 26%"></div><div class="cover-empty" style="width: 74%"></div></div>
	</td>
	<td data-value="26.83" class="pct low">26.83%</td>
	<td data-value="41" class="abs low">11/41</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="10" class="abs low">3/10</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="24" class="abs low">3/24</td>
	<td data-value="26.83" class="pct low">26.83%</td>
	<td data-value="41" class="abs low">11/41</td>
	</tr>

<tr>
	<td class="file low" data-value="Profile.js"><a href="Profile.js.html">Profile.js</a></td>
	<td data-value="35.29" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.29" class="pct low">35.29%</td>
	<td data-value="17" class="abs low">6/17</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="35.29" class="pct low">35.29%</td>
	<td data-value="17" class="abs low">6/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Apr 25 2021 14:56:11 GMT+0530 (India Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    