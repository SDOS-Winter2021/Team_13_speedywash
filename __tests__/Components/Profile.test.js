import React from 'react';
import renderer from 'react-test-renderer';
import Profile from "../../Components/Profile/Profile"
import {test_user} from "../global"

it('Profile renders correctly', () => {
    const tree = renderer.create(<Profile currentUser={test_user} />).toJSON();
    expect(tree).toMatchSnapshot();
})

