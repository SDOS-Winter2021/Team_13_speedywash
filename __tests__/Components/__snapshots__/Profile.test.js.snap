// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<RCTScrollView
  nestedScrollEnabled={false}
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "#DCF2FD",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 17,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "height": "25%",
          "justifyContent": "space-evenly",
          "marginBottom": "8%",
          "marginHorizontal": "4%",
          "marginTop": "5%",
          "marginVertical": "2%",
          "width": "92%",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#006BAD",
              "fontSize": 18,
              "marginHorizontal": "2%",
              "padding": "2%",
              "textAlign": "left",
              "textAlignVertical": "center",
            }
          }
        >
          test
        </Text>
        <Text
          style={
            Object {
              "color": "#006BAD",
              "fontSize": 18,
              "marginHorizontal": "2%",
              "padding": "2%",
              "textAlign": "left",
              "textAlignVertical": "center",
            }
          }
        >
          +919958820305
        </Text>
      </View>
      <Image
        Mode="contain"
        resize={true}
        source={
          Object {
            "uri": "https://firebasestorage.googleapis.com/v0/b/speedywash-7d67a.appspot.com/o/profile%2FxD1ncicw8lbnZGk9sTxPH9yCEi13?alt=media&token=85833276-6e3d-4a86-bbab-1d18c643d48a",
          }
        }
        style={
          Object {
            "borderColor": "#006BAD",
            "borderRadius": 75,
            "borderWidth": 3,
            "height": 120,
            "overflow": "hidden",
            "width": 120,
          }
        }
      />
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "flex": 1,
          "marginBottom": "20%",
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 5,
            "backgroundColor": "white",
            "borderRadius": 7,
            "height": undefined,
            "marginHorizontal": "4%",
            "marginVertical": "1.5%",
            "width": "92%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "margin": "10%",
                }
              }
            >
              Edit Profile
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "aspectRatio": 5,
            "backgroundColor": "white",
            "borderRadius": 7,
            "height": undefined,
            "marginHorizontal": "4%",
            "marginVertical": "1.5%",
            "width": "92%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "margin": "10%",
                }
              }
            >
              My Orders
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "aspectRatio": 5,
            "backgroundColor": "white",
            "borderRadius": 7,
            "height": undefined,
            "marginHorizontal": "4%",
            "marginVertical": "1.5%",
            "width": "92%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "margin": "10%",
                }
              }
            >
              Manage Address
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "aspectRatio": 5,
            "backgroundColor": "white",
            "borderRadius": 7,
            "height": undefined,
            "marginHorizontal": "4%",
            "marginVertical": "1.5%",
            "width": "92%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "margin": "10%",
                }
              }
            >
              Sign Out
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
